src/store/actions.ts

```diff
@@ -50,25 +52,85 @@ export type InitParams = {
 	lexicalCategory?: SearchedItemOption;
 };
 
+interface ValidLexemeData {
+	validLemma: string;
+	validLanguageId: string;
+	validLexicalCategoryId: string;
+	validSpellingVariant: string;
+}
+
 export default function createActions(
 	lexemeCreator: LexemeCreator,
 	langCodeRetriever: LangCodeRetriever,
 	languageCodesProvider: LanguageCodesProvider,
 	tracker: Tracker,
 ): RootActions {
 	return {
-		async [ CREATE_LEXEME ]( { state, commit }: RootContext ): Promise<string> {
-			if ( !state.language || !state.lexicalCategory ) {
-				throw new Error( 'No language or lexical category!' ); // TODO
+		[ GET_VALID_INPUTS ]( { state, commit }: RootContext ): ValidLexemeData {
+			const formData: Partial<ValidLexemeData> = {};
+
+			if ( !state.lemma ) {
+				commit(
+					ADD_PER_FIELD_ERROR,
+					{ field: 'lemmaErrors', error: { messageKey: 'wikibaselexeme-newlexeme-error-no-lemma' } },
+				);
+			} else {
+				formData.validLemma = state.lemma;
+			}
+			if ( !state.language ) {
+				commit(
+					ADD_PER_FIELD_ERROR,
+					{ field: 'languageErrors', error: { messageKey: 'wikibaselexeme-newlexeme-error-no-language' } },
+				);
+			} else {
+				formData.validLanguageId = state.language.id;
+			}
+			if ( !state.lexicalCategory ) {
+				commit(
+					ADD_PER_FIELD_ERROR,
+					{ field: 'lexicalCategoryErrors', error: { messageKey: 'wikibaselexeme-newlexeme-error-no-lexical-category' } },
+				);
+			} else {
+				formData.validLexicalCategoryId = state.lexicalCategory.id;
+			}
+			if ( state.language ) {
+				if ( !state.languageCodeFromLanguageItem && !state.spellingVariant ) {
+					commit(
+						ADD_PER_FIELD_ERROR,
+						{ field: 'spellingVariantErrors', error: { messageKey: 'wikibaselexeme-newlexeme-error-no-spelling-variant' } },
+					);
+				} else {
+					formData.validSpellingVariant = state.spellingVariant ||
+						state.languageCodeFromLanguageItem as string;
```

Comment:
Not sure if it is possible to avoid that type cast here

Reply:
Pushed a commit avoiding it, feel free to revert if you don’t like i.